---
title: "CAME 2024"
date: 2024-09-01
excerpt: "Instructions for CAME 2024"
category: hidden
---


VirtualBox
==========

The Virtual Machine is available from USB sticks, or for download here:
[https://disc-genomics.uibk.ac.at/data/CAME2024_v5.ova](https://disc-genomics.uibk.ac.at/data/CAME2024_v5.ova)

The user is **vmuser** and the password is **vmuser**.

16S Ribosomal RNA Gene Amplicons
================================

## 0. Download the data

```
wget 'https://disc-genomics.uibk.ac.at/data/CAME_SSU.tar.gz' -O - | tar -zx
cd VM_CAME
```

## 1. Download the LTP database

```
wget 'https://disc-genomics.uibk.ac.at/data/LTP_10_2024.arb'
```


Genome-Resolved Metagenomics
============================

## 0. Download the data

```
wget 'https://disc-genomics.uibk.ac.at/data/came24.tar.gz' -O - | tar -zx
cd SLMU
```

You can start by running Nonpareil locally:

```
mkdir -p 00_nonpareil
nonpareil -T kmer -s SLMU.1.fastq.gz -f fastq -b 00_nonpareil/SLMU
NonpareilCurve.R --pdf 00_nonpareil/SLMU.pdf --no-observed 00_nonpareil/SLUM.npo
open 00_nonpareil/SLMU.pdf
```

## 1. Assembly

```
# IMPORTANT: This step might use significant RAM and take a very long time
spades.py --meta -1 SLMU.1.fastq.gz -2 SLMU.2.fastq.gz -o 01_assembly
```

If this is taking too long, you can kill the process by using Ctrl + C,
and instead copy the scaffolds we provide:

```
rm -rf 01_assembly
mkdir -p 01_assembly
cp SLMU.scaffolds.fasta 01_assembly/scaffolds.fasta
```


## 2. Mapping reads to the assembly

```
mkdir -p 02_mapping
bowtie2-build 01_assembly/scaffolds.fasta 02_mapping/SLMU.idx
bowtie2 -1 SLMU.1.fastq.gz -2 SLMU.2.fastq.gz \
      -S 02_mapping/SLMU.sam -x 02_mapping/SLMU.idx --no-unal
samtools view -b 02_mapping/SLMU.sam | samtools sort -o 02_mapping/SLMU.bam -
ls 02_mapping
```


## 3. Binning

```
mkdir -p 03_binning
jgi_summarize_bam_contig_depths --outputDepth 03_binning/SLMU.abund \
      02_mapping/SLMU.bam
metabat2 -i 01_assembly/scaffolds.fasta -a 03_binning/SLMU.abund \
      -o 03_binning/SLMU_bin
ls 03_binning/SLMU_bin.*.fa
```


## 4. Comparing a MAG against a collection of close relatives

This step is optional, and it's intended to demonstrate the capabilities
of MiGA in the command line. Feel free to skip it.

```
miga new -P 04_classification/Sal -t genomes
miga gtdb_get -P 04_classification/Sal -T g__Salinibacter --ref -v
miga add -P 04_classification/Sal -i assembly -t popgenome \
      03_binning/SLMU_bin.*.fa
miga index_wf -o 04_classification/Sal -v
```

That last command will take some time, but you can continue to
part 5 and revisit the results from MiGA later.

Once the run is complete, you can open the file:

```
04_classification/Sal/index.html
```


## 5. Mapping reads to the genome(s)

```
mkdir -p 05_genome_mapping
bowtie2-build 03_binning/SLMU_bin.5.fa 05_genome_mapping/SLMU_bin5.idx
bowtie2 -1 SLMU.1.fastq.gz -2 SLMU.2.fastq.gz \
      -S 05_genome_mapping/SLMU_bin5.sam \
      -x 05_genome_mapping/SLMU_bin5.idx --no-unal
```


## 6. Recruitment plots

```
mkdir -p 06_recplot
rpe build -d 06_recplot/SLMU_5.db -r 05_genome_mapping/SLMU_bin5.sam \
    -g 03_binning/SLMU_bin.5.fa --mag
rpe plot -d 06_recplot/SLMU_5.db
mv recruitment_plots 06_recplot/
```

Open the file manager and open the following file to explore some of the results
locally:

```
06_recplot/recruitment_plots/SLMU_bin5_sam/SLMU_bin_recruitment_plot.html
```


Software Used
=============

- [ARB](http://www.arb-home.de/)
- [Qiime2](https://docs.qiime2.org/)
- [Nonpareil](https://github.com/lmrodriguezr/nonpareil)
- [SPADES](https://github.com/ablab/spades)
- [Bowtie2](https://github.com/BenLangmead/bowtie2)
- [Metabat2](https://bitbucket.org/berkeleylab/metabat)
- [MiGA](https://github.com/bio-miga/miga)
- [RecruitPlotEasy](https://github.com/KGerhardt/RecruitPlotEasy)

And lots of Software used by those above!

